### PR TITLE
refactor javassist compiler: extract class CtClassBuilder

### PR DESCRIPTION
**Pull request by wanghbxxxx**
## What is the purpose of the change

refactor javassist compiler: extract class CtClassBuilder
so the compile process is much more clearer

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/ClassUtils.java [hunk 1]
```diff
@@ -431,5 +431,13 @@ public static String getInitCode(Class<?> type) {
         }
         return map;
     }
+    
+    /**
+     * get classname from qualified name
+     */
+    public static String getSimpleClassName(String qualifiedName) {
+        int i = qualifiedName.lastIndexOf('.');
```
Comment (khanimteyaz) [inline dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/ClassUtils.java]: Better to check and have check for non empty or null string 

`org.apache.dubbo.common.utils.Assert.notEmptyString(qualifiedName, "Name can't be null or blank");`

File: dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/JavassistClassInfo.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+import javassist.CtNewConstructor;
+import javassist.CtNewMethod;
+import javassist.NotFoundException;
+
+/**
+ * Javassist class info contains all the information used for JavassistCompiler, including:
+ * <p>
+ * class name, imported packages, super class name, implemented interfaces, constructors, fields, methods.
+ */
+public class JavassistClassInfo {
```
Comment (khanimteyaz) [inline dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/JavassistClassInfo.java]: As once this class gets initialized then it is not getting modified, I would recommend to use builder for this and then have only getter or make it immutable. 

have addInterface, addConstructor keep it the door open for accidental bug. What do you say?

File: dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/ClassUtils.java (hunk 1 above)
Comment (wanghbxxxx) [reply dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/ClassUtils.java]: thank you for your advice

Comment (wanghbxxxx) [reply dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/ClassUtils.java]: committed

File: dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/JavassistClassInfo.java (hunk 2 above)
Comment (wanghbxxxx) [reply dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/JavassistClassInfo.java]: could you give more detailed information what shall i do, could be some pseudo code or java code.
thank you

Comment (khanimteyaz) [reply dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/JavassistClassInfo.java]: it was saying having setter for collections e.g. setImports(List<String> imports), setInterfaces(List<String> ifaces) .
but looking at again it seems you are doing more than just setter. In this case I think it is ok to have this.

Comment (wanghbxxxx) [reply dubbo-common/src/main/java/org/apache/dubbo/common/compiler/support/JavassistClassInfo.java]: thanks

**Comment by beiwei30** [comment]
LGTM.

**Pull request merged**